### PR TITLE
Int b 21373 validation submit move details task order with files

### PR DESCRIPTION
**Pull request by samaysofo**
## Summary
This pr is to add disabling of the "Submit move details" and "Approve selected" buttons on the Move details page. It also adds and alert message /disabling for "Manage orders" /"Done buttons if there's no uploaded documents. A scroll to view is also added to focus on required fields when data is not entered into a required field. 
Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the application
2. Login as a SC 
3. Search for a Move (could be new or already created), navigate to the Move details page.
4. Scroll down to "Orders" section and click the "View and edit orders" button.
5. "Manage orders" and "Done" buttons should be disabled with a message "File upload is required" if there's no document uploaded. 
6. Click on the "Save" button, should stay on the same page and receive "Please select one or more files"
7. Upload a document and click the Save button.
8. Should be navigated to the "Move details" page with "Submit move details" button enabled (if all required orders fields are entered).
9. Verify disabling of the Submit move details button by repeating step 4 and then deleting the uploaded document and click the "Cancel" button. When navigated back to the Move details page the Submit move details button should be disabled.
10. Login as a TOO
11. Repeat step 3 - 7
12. Should be navigated to the "Move details" page, select a shipment(Move management seems to be selected by default) and the "Approve selected" button should be enabled. 
13. Verify disabling of the Approve selected button by repeating step 4 and then deleting the uploaded document and click the "Cancel" button.
14. Repeat step 12 and the "Approve selected button should be disabled.
15. Login as SC/TOO
16. Repeat step 3 and 4
17. Make sure some of the required fields are not filled out(e.g Date issued, Report by date, Department indicator, Orders number) and scroll out of focus any of the fields not entered.
18. Click on the "Save" button, should scroll to focus the first required field not entered. 

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots

https://github.com/user-attachments/assets/7e6a70d3-fc1c-4480-9b05-40fb131d1f06

https://github.com/user-attachments/assets/11e8831b-d502-4304-a2bc-0bbb7bbfd779

https://github.com/user-attachments/assets/b0d47815-b62e-4145-9ff9-8c52873543a0

### CODE REVIEW

File: src/components/DocumentViewerFileManager/DocumentViewerFileManager.jsx [hunk 1]
```diff
@@ -69,7 +70,10 @@ const DocumentViewerFileManager = ({
       setShowUpload(true);
       setIsExpandedView(true);
     }
-  }, [documentType]);
+    if (required) {
```
Comment (ryan-mchugh) [inline src/components/DocumentViewerFileManager/DocumentViewerFileManager.jsx:73]: (minor) uploadRequired (or something similar)

File: src/components/DocumentViewerFileManager/DocumentViewerFileManager.jsx [hunk 2]
```diff
@@ -28,6 +28,7 @@ const DocumentViewerFileManager = ({
   files,
   documentType,
   updateAmendedDocument,
+  fileUploadrequired,
```
Comment (antgmann) [inline src/components/DocumentViewerFileManager/DocumentViewerFileManager.jsx:31]: Could this be fileUploadRequired instead?

Comment (samaysofo) [reply src/components/DocumentViewerFileManager/DocumentViewerFileManager.jsx]: updated.

Comment (antgmann) [reply src/components/DocumentViewerFileManager/DocumentViewerFileManager.jsx]: Thanks!

File: src/components/DocumentViewerFileManager/DocumentViewerFileManager.test.jsx [hunk 3]
```diff
@@ -96,6 +96,14 @@ describe('DocumentViewerFileManager', () => {
     updateAmendedDocument: jest.fn(),
   };
 
+  const ordersPropsNoFile = {
+    move: { id: 'move-id', locator: 'move-locator' },
+    orderId: 'order-id',
+    fileUploadrequired: true,
```
Comment (ryan-mchugh) [inline src/components/DocumentViewerFileManager/DocumentViewerFileManager.test.jsx]: Need to update props

**Review by ryan-mchugh — APPROVED** [review]
Tested locally and looks good!

**Comment by antgmann** [comment]
So far, everything is looking great. I have noticed some ANDI flags on this page that I may need to open an issue for, but it's unrelated to this PR's changes:
Pasted Graphic 1
Pasted Graphic
Pasted Graphic 2

**Comment by antgmann** [comment]
image
Another issue unrelated to this PR's changes, these console errors are going to become annoying. Andi doesn't like it either. Separate issue inbound

**Review by antgmann — APPROVED** [review]
Everything looks good and functions as expected, this page has a lot of unrelated messiness that would be nice to clean up in separate issues.

**Comment by danieljordan-caci** [comment]
> So far, everything is looking great. I have noticed some ANDI flags on this page that I may need to open an issue for, but it's unrelated to this PR's changes: Pasted Graphic 1 Pasted Graphic Pasted Graphic 2

We do have a handy [feature](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=E-06194) that I have been throwing UI changes into. I think it would be appropriate to throw in a blurb in there about fixing these 508/ANDI errors and hopefully get them fixed in the near future.

**Review by msaki-caci — APPROVED** [review]
LGTM

**Pull request merged**